internal/controller/bmc_controller.go

```diff
@@ -88,12 +91,17 @@ func (r *BMCReconciler) reconcile(ctx context.Context, log logr.Logger, bmcObj *
 		log.V(1).Info("Skipped BMC reconciliation")
 		return ctrl.Result{}, nil
 	}
+	if modified, err := r.handleAnnotionOperations(ctx, log, bmcObj); err != nil || modified {
+		return ctrl.Result{}, err
+	}
 
 	bmcClient, err := bmcutils.GetBMCClientFromBMC(ctx, r.Client, bmcObj, r.Insecure, r.BMCPollingOptions)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to create BMC client: %w", err)
```

Comment:
Maybe we don't want to reset a BMC if e.g. the error thrown due to wrong credentials? Can we check for authentication errors?

Reply:
Looks like we are getting a `http.Response` from the connection request. So we should filter our 403s and related responses and only do a reset in case of 5xx or timeout errors.

Reply:
ok, had to remove some of the error wrappings to be able to read the http status code from the gofish client error